arb_os/errorHandler.mini

```diff
@@ -34,22 +38,32 @@ public impure func errorHandler() {
     // That's OK, because the only "harm" done is that the double-error situation of application error plus ArbGas underflow
     //      gets detected as just an ArbGas underflow.  The effect on the application will be the same either way.
 
-    if (asm() uint { getgas } > (~0 - 300)) {
-        // application ran out of ArbGas
-        asm(665,) { debugprint };
-    } else {
-        // some other error
+
+    if (inErrorHandler) {
         asm(666,) { debugprint };
+    } else {
+        inErrorHandler = true;
+        if (asm() uint { getgas } > (~0 - 500)) {
+            // application ran out of ArbGas
+            let (txid, addrs, parentAddrs) = evmCallStack_stackInfo();
+            asm((665, txid, addrs, parentAddrs),) { debugprint };
+        } else {
+            // some other error
+            let (txid, addrs, parentAddrs) = evmCallStack_stackInfo();
+            asm((666, txid, addrs, parentAddrs),) { debugprint };
+        }
```

Comment:
Probably better to lift the `let (txid, addrs, parentAddrs) = evmCallStack_stackInfo();` out of the if-else to reduce code duplication unless I'm missing something

Reply:
This is deliberate. The condition on the if statement depends on the amount of ArbGas used. I wanted to call evmCallStack_stackInfo after checking the gas condition, rather than before. That way we don't have to worry about whether the ...stackInfo call changes the result of the gas comparison.

Reply:
Ahhh. That makes a ton of sense 😁 

It'd probably be good to add a comment to the code explaining that reasoning so that somebody in the future doesn't accidentally break it

Reply:
Done.